src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -225,7 +225,17 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
   }
 
   function shouldRenderDatePicker(): boolean {
-    return Boolean(props.showDatePicker ?? true) ?? (props.showAutoRefreshOnly && true);
+    return (
+      Boolean(props.showDatePicker ?? true) &&
```

Comment:
nit: ouch to the  3 things that can control the date picker from rendering but we call it three different things. 

we should consider field consistency and if the language and dataset type has the ability to configure the date picker is there. we should be asking the query string manager if we should show date picker. since dataset service and  language service are children of the query string manager . instead accessing the querystring than the services.

the querystring manager could have a helper function like

`supportsTimeFilter(query: Query)` then it accessing the services and return back if that query supports time filters. would make it a little bit easier to handle the logic and a lower lvl

Reply:
not a blocker for this pr